### PR TITLE
Update contributing guide for sample multiple versions

### PR DESCRIPTION
**Pull request by thepetk**
### What does this PR do?:

After the merge of the [PR for the support of branches per version of devfile sample](https://github.com/devfile/registry-support/pull/178), it would be useful to add more information about this to the contributing guide.

### Which issue(s) this PR fixes:

Fixes https://github.com/devfile/api/issues/1209

### PR acceptance criteria:

- [x] Contributing guide
_Have you read the [devfile registry contributing guide](https://github.com/devfile/registry/blob/main/CONTRIBUTING.md) and followed its instructions?_
- [x] Test automation
_Does this repository's tests pass with your changes?_
- [x] Documentation
_Does any documentation need to be updated with your changes?_
- [x] Check Tools Provider
_Have you tested the changes with existing tools, i.e. Odo, Che, Console? (See [devfile registry contributing guide](https://github.com/devfile/registry/blob/main/CONTRIBUTING.md) on how to test changes)_

### How to test changes / Special notes to the reviewer:

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -88,6 +88,45 @@ The devfile samples used in this devfile registry are stored in the `extraDevfil
 4) Fill in the fields in the angle brackets based on your sample. Note that there must be only one git remote for the devfile sample.
 5) Open a pull request against this repository with your changes.
 
+### Adding a new version
+
+In case you want to add another version to a new devfile sample you can update the existing sample inside the `extraDevfileEntries.yaml` file:
+
+1) Verify your Sample functions with Che.
+  
+    - Opening the URL `https://workspaces.openshift.com/#<repository_url>` in your browser should start a workspace where `exec` commands run successfully. Type `task+<space>` in the IDE command palette to see the list of available commands.
```
Comment (maysunfaisal) [inline CONTRIBUTING.md:97]: What does this test?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -88,6 +88,45 @@ The devfile samples used in this devfile registry are stored in the `extraDevfil
...
+        - version: <version1>
+            schemaVersion: <devfile-schemaVersion>
+            git:
+            checkoutFrom:
+                revision: <sample-git-repo-commit-id-or-branch>
+            remotes:
+                origin: <link-to-sample-git-repository>
+    ```
+
+4) Fill in the fields in the angle brackets based on your sample. Note that there must be only one git remote for the devfile sample.
```
Comment (maysunfaisal) [inline CONTRIBUTING.md:127]: but the `origin` remote can point to different URLs for the same sample right?
for example

Go Sample
- Go Sample version Z from X Repo and Y Revision
- Go Sample version C from A Repo and B Revision

File: CONTRIBUTING.md (hunk 1 above)
Comment (thepetk) [reply CONTRIBUTING.md:97]: @maysunfaisal Checks if your sample work. I've tried this URL https://github.com/thepetk/devfile-sample-go-basic/tree/update_devfile in order to check the sample and build it in workspaces.

Maybe I should add more explicit info about branches?

File: CONTRIBUTING.md (hunk 2 above)
Comment (thepetk) [reply CONTRIBUTING.md:129]: I've added a comment in order to point out that using different repos is possible.

**Review by maysunfaisal — APPROVED** [review]

**Comment by maysunfaisal** [comment]
@thepetk needs rebase i think?

**Comment by thepetk** [comment]
> @thepetk needs rebase i think?

@maysunfaisal  done :)

**Review by maysunfaisal — APPROVED** [review]

**Pull request merged**